GitOutTest/Features/Memory/SpanSplitExtensionsTest.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using NUnit.Framework;
+
+namespace GitOut.Features.Memory
+{
+    public class SpanSplitExtensionsTest
+    {
+        [Test]
+        public void SplitShouldSplitSimple()
+        {
+            ReadOnlySpan<char> span = "a b".AsSpan();
+            Range[] result = span.Split();
+            Assert.That(result.Length, Is.EqualTo(2));
+            Assert.That(result[0], Is.EqualTo(0..1));
+            Assert.That(result[1], Is.EqualTo(2..3));
```

Comment:
how does this work?

Reply:
ReadOnlySpan is not allowed to be part of iterator methods (yield basically) or async functions and cannot be returned as a type parameter of generic types (such as IEnumerable). So instead of using that I return a list of Ranges, that can be used to slice parts of the span outside. So instead of a `foreach (string part of full.Split())` is is instead basically `foreach (ReadOnlySpan<char> part of full.Split().Select(range => part[range]))`